api/rs/slint/lib.rs

```diff
@@ -318,6 +318,11 @@ pub mod platform {
     }
 }
 
+#[cfg(any(doc, all(target_os = "android", feature = "backend-android-native-activity")))]
+mod android;
+#[cfg(any(doc, all(target_os = "android", feature = "backend-android-native-activity")))]
+pub use android::*;
```

Comment:
I've put everything inline as that's what we usually do. (as opposed to a public android module)

But the alternative would be to have `slint::android::init`  and re-export under `slint::android::android_activity`.

The advantage is that we could put more documentation in the module docs instead of in the `android_init` function.

But given that there is not much API surface, I think it's fine.

Reply:
I'm surprised by your position as you've always been defending flat namespace in the past :-)

Then you you suggest that the user code look like:
```rust
fn android_main(app: slint::android::android_activity::AndroidApp) {
    slint::android_init(app).unwrap();
```

Ok, `slint::android::android_activity::AndroidApp` could be imported. I think if it is a module, i can re-export it as `slint::android::AndroidApp`.

> I have the sinking feeling that we might have more functionality specific to Android in the future.

That's possible although they may not belong there. Could just be extra function on slint::Window or so. (Could also be extension trait, but the fact you need to import them is not great IMHO)


Regarding the name of the module, I think `android` is simple and obvious.
I don't think `slint::android_activity` is a good name for the module because it is confusing with the crate name (and then it would be `slint::android_activity::android_activity` to access the android_activity crate?)
And i'm not expecting other android backends
Maybe having `slint::backends::android` is fine but do we expect to have more API for backends? I guess we want to export winit at some point.


An alternative to a module is also to make it a different crate:
`slint-android` or something like that.

Reply:
Hah, did I defend that? I thought I was the one who proposed to put the BorrowedOpenGLTextureBuilder into a gpu module.:)

I agree that we don't need android_activity in the name, since that'll be the only support crate to use.

Yes, I think that perhaps slint::backends::android (which contains all types needed without an android_activity sub-module) looks better next to slint::backends::winit, as opposed to slint::winit, which I'm unsure about how it should look like. Purely by name it sounds like a straight re-export of winit, but it wouldn't be. So either backends::winit (which has the types from our crate) or slint::backend_winit looks better to me.